### PR TITLE
Move chapter directories out of `src` and into root directory

### PR DESCRIPTION
**Pull request by feihong**
- [x] Reference the new template project at https://github.com/melange-re/melange-for-react-devs-template/
- [x] Move chapter directories out of `src` and into root directory

I set the base to `develop` branch to make this easier to review, but I actually intend to merge it into `main` branch.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Deploy VitePress site to Pages
 
 on:
-  # Runs on pushes targeting the `develop` branch.
+  # Runs on pushes targeting the `main` branch.
```
Comment (jchavarri) [inline .github/workflows/deploy.yml:4]: This has been changed, but not the line 6 below (the actual rule)

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -50,4 +50,5 @@ jobs:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           publish_dir: ./docs/.vitepress/dist
           enable_jekyll: false
+          force_orphan: true
```
Comment (jchavarri) [inline .github/workflows/deploy.yml:53]: Why was this needed?

**Review by jchavarri — COMMENTED** [review]
> Move chapter directories out of src and into root directory

What motivated this change? I think it was better to keep chapters in `src` folder, especially if we will keep adding more chapters in the future.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (feihong) [reply .github/workflows/deploy.yml:4]: Yes, will change before merging to `main`

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (feihong) [reply .github/workflows/deploy.yml:53]: Sorry, forgot to leave a comment about this. I think it just makes `gh-pages` branch a one-commit branch, which I think makes more sense because it's just basically a copy of whatever is in the `dist` folder.

**Review by feihong — COMMENTED** [review]
> What motivated this change? I think it was better to keep chapters in src folder, especially if we will keep adding more chapters in the future.

I couldn't think of a very practical benefit to having the `src` folder, so I just took everything out. But I don't have strong opinion about it, do you prefer I undo that change?

**Comment by jchavarri** [comment]
> I couldn't think of a very practical benefit to having the `src` folder, so I just took everything out. But I don't have strong opinion about it, do you prefer I undo that change?

My main concern is that there are also `public` and `docs`, so when we merge `develop` into `main`, there will be these folders:

```
celsius-converter-exception
celsius-converter-option
counter
docs
numeric-types
public
```

I think it's a bit messy, considering alphabetic ordering.

**Review by jchavarri — APPROVED** [review]

**Comment by feihong** [comment]
> I think it's a bit messy, considering alphabetic ordering.

Got it, will undo the file moves and put chapter dirs back in src
